pkg/apis/certificates/validation/validation.go

```diff
@@ -60,6 +62,24 @@ func ValidateCertificateSigningRequest(csr *certificates.CertificateSigningReque
 	}
 	if csr.Spec.SignerName == nil {
 		allErrs = append(allErrs, field.Required(specPath.Child("signerName"), "signerName must be provided"))
+	} else {
+		// ensure signerName is of the form domain.com/something and up to 571 characters
+		if len(*csr.Spec.SignerName) > 571 {
+			allErrs = append(allErrs, field.TooLong(specPath.Child("signerName"), *csr.Spec.SignerName, 571))
+		}
+		parts := strings.Split(*csr.Spec.SignerName, "/")
+		if len(parts) != 2 {
+			allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), *csr.Spec.SignerName, "must be a fully qualified domain of the form 'example.com/signer-name'"))
+		} else {
+			// check the domain portion to ensure it is a valid domain label
+			if errs := validation.NameIsDNSSubdomain(parts[0], false); errs != nil {
+				for _, err := range errs {
+					allErrs = append(allErrs, field.Invalid(specPath.Child("signerName"), *csr.Spec.SignerName, err))
+				}
+			}
+
+			// TODO: should we also validate the 'path' component?
```

Comment:
@deads2k @liggitt I was uncertain exactly what validation to perform on the value of `signerName`, but these seemed to make sense. Any suggestions?

Reply:
> @deads2k @liggitt I was uncertain exactly what validation to perform on the value of `signerName`, but these seemed to make sense. Any suggestions?

I favor strictness. There should be something.  I'll throw out. 
 1. length for sure. Needs to be at least one character
 2. I think I'd like to split on dots and verify each bit between the dots was at least one character and `NameIsDNSLabel`

Reply:
I've updated this to check that both the domain and path component have a length of at least 1, and each component split on `.` are DNSLabels (as per `NameIsDNSLabel)`.

Let me know if there are any further restrictions you can think of here. There is no fixed max length per component, but there is a maximum overall length of 571 characters.